airflow-core/src/airflow/dag_processing/manager.py

```diff
@@ -1034,10 +1054,11 @@ def _kill_timed_out_processors(self):
                 )
                 self._file_stats[file] = stat
 
-        # Clean up `self._processors` after iterating over it
+        # Clean up timed-out `self._processors` after iterating over it
         for proc in processors_to_remove:
-            processor = self._processors.pop(proc)
-            processor.logger_filehandle.close()
+            processor = self._processors.pop(proc, None)
+            if processor:
+                processor.logger_filehandle.close()
```

Comment:
```suggestion
            if processor := self._processors.pop(proc, None):
                processor.logger_filehandle.close()
```

There are elsewhere that can benefit from this pattern too.